pmd-java/src/main/java/net/sourceforge/pmd/lang/java/oom/signature/OperationSignature.java

```diff
@@ -67,19 +84,115 @@ public int hashCode() {
     public enum Role {
         GETTER_OR_SETTER, CONSTRUCTOR, METHOD, STATIC;
 
+        private static final Pattern NAME_PATTERN = Pattern.compile("(?:get|set|is|increment|decrement)\\w*");
+
+
         public static Role get(ASTMethodOrConstructorDeclaration node) {
             return node instanceof ASTConstructorDeclaration ? CONSTRUCTOR : get((ASTMethodDeclaration) node);
         }
 
+
         private static Role get(ASTMethodDeclaration node) {
             if (node.isStatic()) {
                 return STATIC;
-            } else if (node.getName() != null && (node.getName().startsWith("get")
-                || node.getName().startsWith("set"))) {
-                return GETTER_OR_SETTER; // TODO:cf better getter or setter detection
+            } else if (isGetterOrSetter(node)) {
+                return GETTER_OR_SETTER;
             } else {
                 return METHOD;
             }
         }
+
+        private static boolean isGetterOrSetter(ASTMethodDeclaration node) {
```

Comment:
Right that might have gotten out of control. I thought looking at the code might yield better recall, but I think precision must be impacted, and the rules are difficult to read. Your approach seems more systematic and less prone to false positives.

I'll refactor that in the next PR